src/plugins/paperplane.js

```diff
@@ -0,0 +1,52 @@
+'use strict'
+
+const web = require('./util/web')
+
+const traceRoute = handler => req => {
+  const { original, route } = req
+
+  if (web.active(original)) {
+    web.enterRoute(original, route)
+  }
+
+  return handler(req)
+}
+
+const wrapMount = (tracer, config) => mount => opts => {
+  const handler = mount(opts)
+
+  const traced = (req, res) =>
+    web.instrument(
+      tracer, config, req, res, 'paperplane.request',
+      () => handler(req, res)
+    )
+
+  return traced
+}
+
+const wrapRoutes = tracer => routes => handlers => {
+  const traced = {}
+
+  for (const route in handlers) {
+    traced[route] = traceRoute(handlers[route])
+  }
+
+  return routes(traced)
+}
+
+function patch (paperplane, tracer, config) {
+  config = web.normalizeConfig(config)
+  this.wrap(paperplane, 'mount', wrapMount(tracer, config))
+  this.wrap(paperplane, 'routes', wrapRoutes(tracer))
+}
+
+function unpatch (paperplane) {
+  this.unwrap(paperplane, ['mount', 'routes'])
+}
+
+module.exports = {
+  name: 'paperplane',
+  versions: ['>=2.3'],
```

Comment:
This version was released only 3 days ago. How difficult would it be to support older versions as well?

Reply:
😅 We actually published that new version to expose the data needed for `Datadog` to do its thing.  

I'm not sure that I could hack in support in anything prior to that, because we didn't expose the original `IncomingMessage` before `2.3`.  Application code only sees POJO's representing the `Request` (details on that [here](https://github.com/articulate/paperplane/blob/master/docs/getting-started.md)), and we didn't pass along the `req.original` until now.

That's important because `dd-trace` [stores pertinent data on the original request](https://github.com/DataDog/dd-trace-js/blob/master/src/plugins/util/web.js#L72-L74), so without a reference to it, we're a bit lost.

We do make an effort to keep our services on the latest `paperplane`, and [we publish docs to make migrating easy](https://github.com/articulate/paperplane/blob/master/docs/migration-guide.md).  So hopefully lack of support for prior versions shouldn't prove an issue.

Reply:
Fair enough. I think it would be possible to support older versions by not using `web.js` and doing the entire instrumentation directly in the `paperplane` plugin, but as you said this may not be necessary, so I'm fine doing it only if we get an explicit request for it.